src/ucp/core/ucp_worker.c

```diff
@@ -887,6 +887,16 @@ static int ucp_worker_iface_support_keepalive(ucp_worker_iface_t *wiface)
                               UCT_IFACE_FLAG_EP_KEEPALIVE);
 }
 
+/* Compare target iface atomic capabilities to candidate's, and decide if candidate is better */
+static int ucp_worker_iface_is_better_atomic_caps(uct_iface_attr_t *cand_attr,
+                                                  uct_iface_attr_t *target_attr)
+{
+    return (ucs_test_all_flags(cand_attr->cap.atomic32.op_flags, target_attr->cap.atomic32.op_flags)) &&
+           (ucs_test_all_flags(cand_attr->cap.atomic32.fop_flags, target_attr->cap.atomic32.fop_flags)) &&
+           (ucs_test_all_flags(cand_attr->cap.atomic64.op_flags, target_attr->cap.atomic64.op_flags)) &&
+           (ucs_test_all_flags(cand_attr->cap.atomic64.fop_flags, target_attr->cap.atomic64.fop_flags));
```

Comment:
Did you test it, at least manually (runtime injection)?

Reply:
I ran ucx_info and verified I that this code runs properly (returns true for subset and false for no subset)